checker/match.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	"fmt"
+	"regexp"
+
+	authz "github.com/envoyproxy/data-plane-api/api/auth"
+
+	"github.com/projectcalico/app-policy/proto"
+	"github.com/projectcalico/libcalico-go/lib/selector"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const SPIFFE_ID_PATTERN = "^spiffe://[^/]+/ns/([^/]+)/sa/([^/]+)$"
+
+var spiffeIdRegExp *regexp.Regexp
+
+// match checks if the Rule matches the request.  It returns true if the Rule matches, false otherwise.
+func match(rule *proto.Rule, req *authz.CheckRequest) bool {
+	log.Debugf("Checking rule %v on request %v", rule, req)
+	attr := req.GetAttributes()
+	return matchSource(rule, attr.GetSource()) && matchRequest(rule, attr.GetRequest())
+}
+
+func matchSource(r *proto.Rule, peer *authz.AttributeContext_Peer) bool {
+	// TODO IPSets
+	return matchServiceAccounts(r.GetSrcServiceAccount(), peer)
+}
+
+func matchRequest(rule *proto.Rule, req *authz.AttributeContext_Request) bool {
+	log.WithFields(log.Fields{
```

Comment:
if it's only one field then you can just do `log.WithField("request", req).Debug...`